log4j-core/src/main/java/org/apache/logging/log4j/core/util/datetime/FixedDateFormat.java

```diff
@@ -255,7 +261,7 @@ private static int[] nanoRange(final String pattern) {
          * @return the length of the resulting formatted date and time strings
          */
         public int getLength() {
-            return pattern.length() - escapeCount;
+            return pattern.length() - escapeCount + extraTimeZoneFormatLength;
```

Comment:
Given this is fixed, do we still need the `char[]`-size fix in `InstantFormatter`?